### PR TITLE
pydantic: add priority to pydantic models

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

CRAFT-1645

### CODE REVIEW

File: tests/unit/repo/test_projects.py [hunk 1]
```diff
@@ -16,17 +16,18 @@
...
-    def test_apt_ppa_valid(self):
-        repo = {
-            "type": "apt",
-            "ppa": "test/somerepo",
-        }
+    @pytest.mark.parametrize(
+        "priority", ["always", "prefer", "defer", 1000, 990, 500, 100, -1, None]
+    )
+    def test_apt_ppa_valid(self, priority):
+        repo = {"type": "apt", "ppa": "test/somerepo", "priority": priority}
```
Comment (tigarmo) [inline tests/unit/repo/test_projects.py:30]: For symmetry with `test_apt_deb_valid()`, how about if priority is `None` we don't add it to the `repo` dict (as opposed to adding a key with `None` value)?

File: craft_archives/repo/projects.py [hunk 2]
```diff
@@ -28,6 +30,8 @@
 else:
     KeyIdStr = constr(regex=r"^[0-9A-F]{40}$")
 
+PriorityValue = Optional[Union[Literal["always", "prefer", "defer"], int]]
```
Comment (tigarmo) [inline craft_archives/repo/projects.py:33]: What's your opinion on `PriorityValue` being an `Optional` as it is here, versus it being the `Union` bit and then the `priority` attribute being an `Optional[PriorityValue]`? At first glance I think I prefer the second alternative, I think it's more consistent with the other attribute in `Apt`

**Review by tigarmo — COMMENTED** [review]
Thanks for the quick PR

File: craft_archives/repo/projects.py (hunk 2 above)
Comment (lengau) [reply craft_archives/repo/projects.py:33]: Agreed. Changing

**Review by tigarmo — APPROVED** [review]
thanks for the quick turnaround!

**Pull request merged**